Products/ZenModel/migrate/editServicesStartUp.py

```diff
@@ -0,0 +1,129 @@
+
+    version = Migrate.Version(SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION) 
+
+    def cutover(self, dmd):
+        try:
+            ctx = sm.ServiceContext()
+        except sm.ServiceMigrationError:
+            log.info("Couldn't generate service context, skipping.")
+            return
+
+        def update_service(collection):
+            services = filter(lambda s: s.name in collection, ctx.services)
+            for service in services:
+                service.startup = command.format(service.name)
+                service.runAs = "zenoss"
+
+        service_names = ("zencommand", "zenmodeler", "zenperfsnmp", "zenping", "zenprocess", "zenstatus")
+        command = "/opt/zenoss/bin/{} run -c --logfileonly --workers {{.Instances}} --workerid $CONTROLPLANE_INSTANCE_ID --monitor {{(parent .).Name}}"
+
+        update_service(service_names)
+
+        service_names = ("zensyslog", "zentrap", "zenhub")
+        command = "/opt/zenoss/bin/{} run -c --logfileonly --monitor {{(parent .).Name}}"
+
+        update_service(service_names)
+
+        svcs = {
+            'zminion': 'zminion',
+            'CentralQuery': 'central-query',
+            'MetricConsumer': 'metric-consumer-app',
+            'MetricShipper': 'metricshipper',
+            'Zauth': 'zauth',
+        }
+
+        command = "/opt/zenoss/bin/supervisord -n -c /opt/zenoss/etc/{}/supervisord.conf"
+
+        services = filter(lambda s: s.name in svcs.keys(), ctx.services)
+        for service in services:
+            service.startup = command.format(svcs[service.name])
+            service.runAs = "zenoss"
+            update_config(service)
+
+        svcs = {
+            'Zope': 'zope',
+            'zenapi': 'zenapi',
+            'zenreports': 'zenreports',
+        }
+
+        command = "/opt/zenoss/bin/runzope"
+
+        services = filter(lambda s: s.name in svcs.keys(), ctx.services)
+        for service in services:
+            service.startup = command
+            env = "CONFIG_FILE=/opt/zenoss/etc/{}.conf".format(svcs[service.name])
+            if not env in service.environment:
+                service.environment.append(env)
+            service.runAs = "zenoss"
+            
+        svcs = {
+            "zenhubworker (adm)": "/opt/zenoss/bin/zenhubworker run -c --logfileonly --monitor {{(parent .).Name}} --workerid $CONTROLPLANE_INSTANCE_ID adm",
+            "zenhubworker (default)": "/opt/zenoss/bin/zenhubworker run -c --logfileonly --monitor {{(parent .).Name}} --workerid $CONTROLPLANE_INSTANCE_ID default",
+            "zenhubiworker": "/opt/zenoss/bin/zenhubiworker run -c --duallog --hub {{(parent .).Name}} --workerid $CONTROLPLANE_INSTANCE_ID",
+            "zeneventd": "/usr/bin/nice -n 10 /opt/zenoss/bin/zeneventd run -c --logfileonly ",
+            "zenactiond": "/opt/zenoss/bin/zenactiond run -c --logfileonly --workerid $CONTROLPLANE_INSTANCE_ID",
+            "zenjobs": "/opt/zenoss/bin/zenjobs run --logfileonly",
+        }
```

Comment:
And in here.

Of course you could use Python's `%` string interpolation to avoid escaping the double `{` fields.